AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -256,47 +300,52 @@ public class StorageBlobClient: PipelineClient, PagedCollectionDelegate {
-                                   queryParams: queryParams,
-                                   headerParams: headerParams)
-        let context: [String: AnyObject] = [
-            ContextKey.allowedStatusCodes.rawValue: [200, 206] as AnyObject,
-        ]
-        run(request: request, context: context, completion: { result, httpResponse in
+    /**
+     Download a blob from a specific container.
+     - Parameter blob: The name of the blob.
+     - Parameter fromContainer: The name of the container.
+     - Parameter withOptions: A `DownloadBlobOptions` object to control the download operation.
+     - Parameter then: An `HttpResultHandler` closure that returns a `StorageStreamDownloader` object on success.
+     */
+    public func download(blob: String, fromContainer container: String, withOptions options: DownloadBlobOptions? = nil,
+                         then completion: @escaping HttpResultHandler<StorageStreamDownloader>) throws {
+        let downloader = try StorageStreamDownloader(client: self, name: blob, container: container, options: options)
+        downloader.initialRequest() { result, httpResponse in
             switch result {
-            case let .success(data):
-                guard let data = data else {
-                    let noDataError = HttpResponseError.decode("Response data expected but not found.")
-                    completion(.failure(noDataError), httpResponse)
-                    return
-                }
-                completion(.success(data), httpResponse)
+            case .success(_):
+                completion(.success(downloader), httpResponse)
             case let .failure(error):
                 completion(.failure(error), httpResponse)
             }
-        })
+        }
+    }
+
+    /**
+     Download a blob from a specific container.
+     - Parameter url: A URL to a blob to download.
+     - Parameter options: A `DownloadBlobOptions` object to control the download operation.
+     - Parameter completion: An `HttpResultHandler` closure that returns a `StorageStreamDownloader` object on success.
+     */
+    public func download(url: URL, withOptions options: DownloadBlobOptions? = nil,
+                         then completion: @escaping HttpResultHandler<StorageStreamDownloader>) throws {
+        let (host, container, blob) = try parse(url: url)
+        // ensure that if the base URL is altered temporarily because of the URL that it is switched back.
+        let originalHost = baseUrl
+        defer { baseUrl = originalHost }
+        baseUrl = host
+        try download(blob: blob, fromContainer: container, withOptions: options, then: completion)
+    }
+
+    /**
+     Create a simple URL for a blob.
+     - Parameter blob: Name of the blob.
+     - Parameter container: Name of the container.
+     - Returns: The URL of the blob.
+     */
+    public func url(forBlob blob: String, inContainer container: String) -> URL? {
+        var urlString = baseUrl
```

Comment:
Is there no URLBuilder or similar thing that lets us add path components? Just always hate seeing manual path manipulation.

Reply:
Agreed. I rewrote this using `URL.appendPathComponent`.